modules/effects/src/main/scala/dev/profunktor/redis4cats/redis.scala

```diff
@@ -42,22 +43,43 @@ import io.lettuce.core.{
   ZAddArgs,
   ZAggregateArgs,
   ZStoreArgs,
+  ExpireArgs => JExpireArgs,
   GetExArgs => JGetExArgs,
   Limit => JLimit,
   Range => JRange,
   ReadFrom => JReadFrom,
   ScanCursor => JScanCursor,
-  SetArgs => JSetArgs,
-  ExpireArgs => JExpireArgs
+  SetArgs => JSetArgs
 }
 import io.lettuce.core.api.async.RedisAsyncCommands
 import io.lettuce.core.cluster.api.async.RedisClusterAsyncCommands
 import io.lettuce.core.cluster.api.sync.{ RedisClusterCommands => RedisClusterSyncCommands }
+import org.typelevel.keypool.KeyPool
 
 import scala.concurrent.duration._
 
 object Redis {
 
+  object Pool {
+    final case class PoolSettings(maxTotal: Int, maxIdle: Int, idleTimeAllowedInPool: FiniteDuration)
+    object PoolSettings {
+      import scala.language.postfixOps
+      object Defaults {
+        val maxTotal: Int                         = Math.max(10, Runtime.getRuntime.availableProcessors())
```

Comment:
This would be better off being an effectful constructor as it's not a constant, e.g.

```scala
object Settings {
  def default[F[_]: Sync]: F[Settings] =
    Sync[F].blocking(Runtime.getRuntime.availableProcessors()).map { cpu =>
      Settings(maxTotal = Math.max(10, cpu), maxIdle = 2, idleTimeAllowedInPool = 60.seconds)
    }
}
```

Which can be used as follows:

```scala
Redis.Pool.Settings.default[IO] // IO[Pool.Settings]
```

Reply:
@gvolpe  so I've taken your suggestions, however this is a breaking change to the interface while I was attempting to maintain backwards compatibility.  Now you need Sync in scope vs MonadThrow was all you needed before.  Also by adding the effect we've had to add another function so you can pass in custom pool settings.  I agree this solution is better however I was trying to maintain backwards compatibility.

Reply:
@gvolpe refactored per your suggestions in 6d1d873